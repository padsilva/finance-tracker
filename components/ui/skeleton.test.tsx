import { render } from "@testing-library/react";

import { Skeleton } from "./skeleton";

describe("Skeleton Component", () => {
  it("should render basic skeleton correctly", () => {
    const { container } = render(<Skeleton />);

    expect(container).toMatchSnapshot();
  });

  it("should render with custom className", () => {
    const { container } = render(
      <Skeleton className="custom-skeleton h-8 w-32" />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with different sizes", () => {
    const { container } = render(
      <>
        <Skeleton className="h-4 w-16" /> {/* Small */}
        <Skeleton className="h-6 w-24" /> {/* Medium */}
        <Skeleton className="h-8 w-32" /> {/* Large */}
        <Skeleton className="h-12 w-48" /> {/* Extra Large */}
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render common UI element skeletons", () => {
    const { container } = render(
      <div className="space-y-4">
        {/* Avatar skeleton */}
        <Skeleton className="h-12 w-12 rounded-full" />

        {/* Text line skeletons */}
        <div className="space-y-2">
          <Skeleton className="h-4 w-3/4" />
          <Skeleton className="h-4 w-1/2" />
          <Skeleton className="h-4 w-5/6" />
        </div>

        {/* Card skeleton */}
        <div className="space-y-3">
          <Skeleton className="h-32 w-full rounded-lg" />
          <Skeleton className="h-4 w-2/3" />
          <Skeleton className="h-4 w-1/3" />
        </div>

        {/* Button skeleton */}
        <Skeleton className="h-9 w-24 rounded-md" />
      </div>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with different shapes", () => {
    const { container } = render(
      <>
        <Skeleton className="h-8 w-8 rounded-full" /> {/* Circle */}
        <Skeleton className="h-8 w-32 rounded-md" /> {/* Rounded */}
        <Skeleton className="h-8 w-32 rounded-none" /> {/* Square */}
        <Skeleton className="h-8 w-32 rounded-full" /> {/* Pill */}
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render loading card layout", () => {
    const { container } = render(
      <div className="space-y-4 rounded-lg border p-6">
        <div className="flex items-center space-x-4">
          <Skeleton className="h-12 w-12 rounded-full" />
          <div className="space-y-2">
            <Skeleton className="h-4 w-32" />
            <Skeleton className="h-4 w-24" />
          </div>
        </div>
        <Skeleton className="h-32 w-full" />
        <div className="space-y-2">
          <Skeleton className="h-4 w-full" />
          <Skeleton className="h-4 w-4/5" />
          <Skeleton className="h-4 w-3/5" />
        </div>
      </div>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render loading table layout", () => {
    const { container } = render(
      <div className="space-y-4">
        <div className="flex items-center justify-between">
          <Skeleton className="h-8 w-32" />
          <Skeleton className="h-8 w-8" />
        </div>
        {Array.from({ length: 5 }).map((_, i) => (
          <div key={i} className="flex items-center space-x-4">
            <Skeleton className="h-12 w-12" />
            <Skeleton className="h-4 w-full" />
          </div>
        ))}
      </div>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render with custom animations", () => {
    const { container } = render(
      <>
        <Skeleton className="animate-pulse" /> {/* Default */}
        <Skeleton className="animate-[pulse_2s_ease-in-out_infinite]" />{" "}
        {/* Custom timing */}
        <Skeleton className="animate-none" /> {/* No animation */}
      </>,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render responsive skeletons", () => {
    const { container } = render(
      <Skeleton className="h-16 w-16 md:h-24 md:w-24 lg:h-32 lg:w-32" />,
    );

    expect(container).toMatchSnapshot();
  });

  it("should render grid layout skeletons", () => {
    const { container } = render(
      <div className="grid grid-cols-3 gap-4">
        {Array.from({ length: 6 }).map((_, i) => (
          <Skeleton key={i} className="h-32 w-full rounded-lg" />
        ))}
      </div>,
    );

    expect(container).toMatchSnapshot();
  });
});
