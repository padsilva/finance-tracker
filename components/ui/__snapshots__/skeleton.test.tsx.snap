// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Skeleton Component should render basic skeleton correctly 1`] = `
<div>
  <div
    class="animate-pulse rounded-md bg-primary/10"
  />
</div>
`;

exports[`Skeleton Component should render common UI element skeletons 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="animate-pulse bg-primary/10 h-12 w-12 rounded-full"
    />
    <div
      class="space-y-2"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-3/4"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-1/2"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-5/6"
      />
    </div>
    <div
      class="space-y-3"
    >
      <div
        class="animate-pulse bg-primary/10 h-32 w-full rounded-lg"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-2/3"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-1/3"
      />
    </div>
    <div
      class="animate-pulse bg-primary/10 h-9 w-24 rounded-md"
    />
  </div>
</div>
`;

exports[`Skeleton Component should render grid layout skeletons 1`] = `
<div>
  <div
    class="grid grid-cols-3 gap-4"
  >
    <div
      class="animate-pulse bg-primary/10 h-32 w-full rounded-lg"
    />
    <div
      class="animate-pulse bg-primary/10 h-32 w-full rounded-lg"
    />
    <div
      class="animate-pulse bg-primary/10 h-32 w-full rounded-lg"
    />
    <div
      class="animate-pulse bg-primary/10 h-32 w-full rounded-lg"
    />
    <div
      class="animate-pulse bg-primary/10 h-32 w-full rounded-lg"
    />
    <div
      class="animate-pulse bg-primary/10 h-32 w-full rounded-lg"
    />
  </div>
</div>
`;

exports[`Skeleton Component should render loading card layout 1`] = `
<div>
  <div
    class="space-y-4 rounded-lg border p-6"
  >
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="animate-pulse bg-primary/10 h-12 w-12 rounded-full"
      />
      <div
        class="space-y-2"
      >
        <div
          class="animate-pulse rounded-md bg-primary/10 h-4 w-32"
        />
        <div
          class="animate-pulse rounded-md bg-primary/10 h-4 w-24"
        />
      </div>
    </div>
    <div
      class="animate-pulse rounded-md bg-primary/10 h-32 w-full"
    />
    <div
      class="space-y-2"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-full"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-4/5"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-3/5"
      />
    </div>
  </div>
</div>
`;

exports[`Skeleton Component should render loading table layout 1`] = `
<div>
  <div
    class="space-y-4"
  >
    <div
      class="flex items-center justify-between"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-8 w-32"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-8 w-8"
      />
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-12 w-12"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-full"
      />
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-12 w-12"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-full"
      />
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-12 w-12"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-full"
      />
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-12 w-12"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-full"
      />
    </div>
    <div
      class="flex items-center space-x-4"
    >
      <div
        class="animate-pulse rounded-md bg-primary/10 h-12 w-12"
      />
      <div
        class="animate-pulse rounded-md bg-primary/10 h-4 w-full"
      />
    </div>
  </div>
</div>
`;

exports[`Skeleton Component should render responsive skeletons 1`] = `
<div>
  <div
    class="animate-pulse rounded-md bg-primary/10 h-16 w-16 md:h-24 md:w-24 lg:h-32 lg:w-32"
  />
</div>
`;

exports[`Skeleton Component should render with custom animations 1`] = `
<div>
  <div
    class="rounded-md bg-primary/10 animate-pulse"
  />
   
  <div
    class="rounded-md bg-primary/10 animate-[pulse_2s_ease-in-out_infinite]"
  />
   
  <div
    class="rounded-md bg-primary/10 animate-none"
  />
   
</div>
`;

exports[`Skeleton Component should render with custom className 1`] = `
<div>
  <div
    class="animate-pulse rounded-md bg-primary/10 custom-skeleton h-8 w-32"
  />
</div>
`;

exports[`Skeleton Component should render with different shapes 1`] = `
<div>
  <div
    class="animate-pulse bg-primary/10 h-8 w-8 rounded-full"
  />
   
  <div
    class="animate-pulse bg-primary/10 h-8 w-32 rounded-md"
  />
   
  <div
    class="animate-pulse bg-primary/10 h-8 w-32 rounded-none"
  />
   
  <div
    class="animate-pulse bg-primary/10 h-8 w-32 rounded-full"
  />
   
</div>
`;

exports[`Skeleton Component should render with different sizes 1`] = `
<div>
  <div
    class="animate-pulse rounded-md bg-primary/10 h-4 w-16"
  />
   
  <div
    class="animate-pulse rounded-md bg-primary/10 h-6 w-24"
  />
   
  <div
    class="animate-pulse rounded-md bg-primary/10 h-8 w-32"
  />
   
  <div
    class="animate-pulse rounded-md bg-primary/10 h-12 w-48"
  />
   
</div>
`;
